### ISSUE TITLE
Empty entry in TOC, when table is added between two H2's

### ISSUE THREAD

**Issue by klartext** [michaelrsweet/htmldoc#503]
With this md-code:

```
# H1

## H2 - first

|Foo  |Bar  |
|-----|-----|
|01   |02   |

## H2 - second

```

I get three entries on the H2-level, two for the H2's and one empty one for the table.

![toc-with empty-entry](https://user-images.githubusercontent.com/3165583/232335926-2a7d6c97-ac39-4ca5-ae86-5f45982c0518.png)

**Comment by michaelrsweet** [comment]
Hmm, will need to look at the markdown parser to see what's up here...

**Comment by michaelrsweet** [comment]
[master cbf3130] Fix markdown parsing bug (Issue #503)
